src/extension.ts

```diff
@@ -67,22 +67,34 @@ export function activate(context: vscode.ExtensionContext) {
 			);
 
 			lspClient.onReady().then(() => {
-				lspClient.onNotification('caNotification', (data) => {
+				lspClient.onNotification('caNotification', (respData) => {
 					vscode.window.withProgress({ location: vscode.ProgressLocation.Window, title: StatusMessages.EXT_TITLE}, p => {
 						return new Promise((resolve, reject) => {
-							p.report({message: 'Analyzing dependencies for any security vulnerability' });
-							p.report({message: data });
+							p.report({message: 'Checking for security vulnerabilities ...' });
+							p.report({message: respData.data });
 							setTimeout(function () {	
 							  resolve();
-							}, 2000);
+							  if(respData && respData.hasOwnProperty('isEditAction') && !respData.isEditAction) {
+								showInfoOnfileOpen(respData.data);
+							  }
+							}, 2500);

```

Comment:
why we are using setTimeout here?

Reply:
setTimeout is used here with an interval of 2.5 secs to resolve progress notification send from Language Server, if it won't be used user won't be able to see any progress as API calls via LS are pretty quick.